core/src/apps/bitcoin/keychain.py

```diff
@@ -234,23 +243,44 @@ def get_coin_by_name(coin_name: str | None) -> coininfo.CoinInfo:
 
 
 async def get_keychain_for_coin(
-    ctx: wire.Context, coin_name: str | None
+    ctx: wire.Context, coin_name: str | None, unlock_pattern: str | None = None
 ) -> tuple[Keychain, coininfo.CoinInfo]:
     coin = get_coin_by_name(coin_name)
-    schemas = get_schemas_for_coin(coin)
+    schemas = get_schemas_for_coin(coin, unlock_pattern)
     slip21_namespaces = [[b"SLIP-0019"], [b"SLIP-0024"]]
     keychain = await get_keychain(ctx, coin.curve_name, schemas, slip21_namespaces)
     return keychain, coin
 
 
+def _get_unlock_pattern(
+    msg: MessageType, auth_msg: MessageType | None = None
+) -> str | None:
+    if AuthorizeCoinJoin.is_type_of(msg):
+        return PATTERN_SLIP25_TAPROOT
+
+    if (
+        auth_msg is not None
+        and (AuthorizeCoinJoin.is_type_of(auth_msg) or UnlockPath.is_type_of(auth_msg))
```

Comment:
fwiw you're not calling `check_get_public_key` here so hypothetically speaking the preauthorization could be using some other path.

maybe change `check_get_public_key` into `check_path_access` or something, and then reuse the function that I proposed for `get_public_key` here as well?

Reply:
Something like this: https://github.com/trezor/trezor-firmware/pull/2289/commits/9c7d2307c1b74af6f78683f042411ccbe3513652

Reply:
I pushed some more changes to improve the commentary https://github.com/trezor/trezor-firmware/pull/2289/commits/e0803c17fa2cdb54d1bb17505fb3dfab48f2e127.

Reply:
https://github.com/trezor/trezor-firmware/pull/2289/commits/47a80169b7d09af48410b4f9a53a843536380a99: I reverted the changes to `PathSchema.match()` made in https://github.com/trezor/trezor-firmware/commit/9c7d2307c1b74af6f78683f042411ccbe3513652 and instead introduced `PathSchema.restrict()`.